src/lib/FreeC/LiftedIR/Syntax/Type.hs

```diff
@@ -44,3 +44,23 @@ data Type
       , typeFreeArg :: Type
       }
  deriving (Eq, Show)
+
+-------------------------------------------------------------------------------
+-- Utility                                                                   --
+-------------------------------------------------------------------------------
+
+-- | Applies a lifted IR type constructor to a type level expression.
+typeApp :: Type -> Type -> Type
+typeApp (TypeCon srcSpan name ts dec) t = TypeCon srcSpan name (t : ts) dec
+typeApp _ _ = error "Cannot apply type to non type constructor!"
```

Comment:
The problem I've got with this function is that `typeApp (D t₁ … tₙ) t₀ = D t₀ t₁ … tₙ`. That's why `LIR.convertType'` reverses the argument list and why you have to apply `reverse` in `Agda.convertType`. The reversing of arguments is a potential source of errors and not documented anywhere. I think that `LIR.convertType'` should be implemented with an accumulator internally such that the order of arguments is preserved.